tools/hisat/hisat.xml

```diff
@@ -0,0 +1,344 @@
+    <command><![CDATA[
+        #if str($spliced_options.spliced_options_selector) == "advanced" and str($spliced_options.known_splice_gff) != 'None':
+            ln -s "${spliced_options.known_splice_gff}" splice_sites.gff &&
+            python $__tool_directory__/extract_splice_sites.py splice_sites.gff > splice_sites.txt &&
+        #end if
+        #if $reference_genome.reference_genome_source == "history":
+            ln -s "$reference_genome.history_item" genome.fa &&
+            hisat-build genome.fa genome && 
+            #set index_path = 'genome'
+        #else:
+            #set index_path = $reference_genome.index.fields.path
+        #end if
+        hisat -p \${GALAXY_SLOTS:-1} -x "${index_path}"
+        #if str($input_format.paired.paired_selector) == 'paired':
+            -1 "${reads_f}" -2 "${reads_r}"
+        #else if str($input_format.paired.paired_selector) == 'paired_collection':
+            -1 "${input_format.paired.reads.forward}" -2 "${input_format.paired.reads.reverse}"
+        #else if str($input_format.paired.paired_selector) == 'paired_list':
+            #set forward_reads = []
+            #set reverse_reads = []
+            #for read in $input_format.paired.reads:
+                $forward_reads.append($read.forward)
+                $reverse_reads.append($read.reverse)
+            #end for
+            -1 "${','.join([str($read) for read in $forward_reads])}"
+            -2 "${','.join([str($read) for read in $reverse_reads])}"
+        #else:
+            -U "${reads}"
+        #end if
+        #if $input_format.input_format_selector == 'fasta':
+            -f
+        #end if
+        #if str($input_options.input_options_selector) == "advanced":
+            -s ${skip} -u ${stop_after} -5 ${trim_five} -3 ${trim_three}
+        #end if
+        #if str($scoring_options.scoring_options_selector) == "advanced":
+            --ma ${match_bonus} --mp ${max_mismatch},${min_mismatch}
+            --np ${ambiguous_penalty} --rdg ${read_open_penalty},${read_extend_penalty} --rfg ${ref_open_penalty},${ref_extend_penalty}
+        #end if
+        #if str($alignment_options.alignment_options_selector) == "advanced":
+            --n-ceil ${alignment_options.function_type},${alignment_options.constant_term},${alignment_options.coefficient_term}
+            ${skip_fw} ${skip_rc}
+        #end if
+        #if str($spliced_options.spliced_options_selector) == "advanced":
+            --pen-cansplice ${spliced_options.canonical_penalty} --pen-noncansplice ${spliced_options.noncanonical_penalty}
+            --pen-intronlen ${spliced_options.function_type},${spliced_options.constant_term},${spliced_options.coefficient_term}
+            #if str($spliced_options.known_splice_gff) != 'None':
+                --known-splicesite-infile splice_sites.txt
+            #end if
+        #end if
+        -S alignments.sam &&
+        samtools view -bS alignments.sam -o alignments.bam &&
```

Comment:
Can we improve this to save IO, using a pipe and the `-o` option? Moreover, if we require some recent version of samtools we can use multiple threads.